tests/ballerina-integration-test/src/test/java/org/ballerinalang/test/socket/SocketClientCallbackServiceTestCase.java

```diff
@@ -85,4 +85,26 @@ public void testSocketServerJoinLeave() throws BallerinaTestException {
         joineeServerLeecher.waitForText(20000);
         serverInstance.removeAllLeechers();
     }
+
+    @Test(description = "Check read timeout")
+    public void testSocketReadTimeout() {
+        LogLeecher timeoutLeecher = new LogLeecher("Read timed out");
+        serverInstance.addLogLeecher(timeoutLeecher);
+        try (SocketChannel socketChannel = SocketChannel.open()) {
+            socketChannel.configureBlocking(true);
+            socketChannel.connect(new InetSocketAddress("localhost", 61599));
+            ByteBuffer buf = ByteBuffer.allocate(64);
+            String welcomeMsg = "Hello Ballerina";
+            buf.put(welcomeMsg.getBytes(StandardCharsets.UTF_8));
+            buf.flip();
+            while (buf.hasRemaining()) {
+                socketChannel.write(buf);
+            }
+            Thread.sleep(2000L);
+            timeoutLeecher.waitForText(22000);
```

Comment:
Instead of having a thread sleep can't we only use the leecher timeout?

Reply:
The read will timeout after 20 seconds in here. Added thread.sleep add some grace period to prevent intermittent test failures when the server get busy due to some other factors.